### PR TITLE
Fix issue of session termination

### PR DESCRIPTION
**Pull request by GANGANI**
Fix issue of session termination.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 1]
```diff
@@ -57,17 +61,71 @@ public void terminateSessionsOfUser(String username, String userStoreDomain, Str
             UserSessionException {
 
         validate(username, userStoreDomain, tenantDomain);
-        List<String> sessionListOfUser = getSessionsOfUser(username, userStoreDomain, tenantDomain);
 
-        if (!sessionListOfUser.isEmpty()) {
-            if (log.isDebugEnabled()) {
-                log.debug("Terminating all the active sessions of user: " + username + " of userstore domain: " +
-                        userStoreDomain + " in tenant: " + tenantDomain);
+        String userId = resolveUserIdFromUsername(getTenantId(tenantDomain), userStoreDomain, username);
```
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:65]: With the change, we have removed the debug log that was previously there. Is there any specific reason to do that?

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 2]
```diff
@@ -57,17 +61,71 @@ public void terminateSessionsOfUser(String username, String userStoreDomain, Str
...
+        try {
+            terminateSessionsByUserId(userId);
+        } catch (SessionManagementException e) {
+            throw new UserSessionException("Error while terminating sessions of user.", e);
```
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:69]: Include some context information like full qualified username or userid

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java [hunk 3]
```diff
@@ -57,17 +61,71 @@ public void terminateSessionsOfUser(String username, String userStoreDomain, Str
...
+            if (userStoreDomain == null) {
+                userStoreDomain = UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME;
+            }
+            UserStoreManager userStoreManager = getUserStoreManager(tenantId, userStoreDomain);
+            try {
+                if (userStoreManager instanceof AbstractUserStoreManager) {
+                    return ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(username);
+                }
+                if (log.isDebugEnabled()) {
+                    log.debug("Provided user store manager for the user: " + username + ", is not an instance of the " +
```
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:95]: Would be better to have the userstoreDomain included as a debug log

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java (hunk 1 above)
Comment (GANGANI) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:65]: Fixed 8a7f2322bf7980e1b678bc34667f82699d5d1ce1

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java (hunk 2 above)
Comment (GANGANI) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:69]: Fixed 8a7f2322bf7980e1b678bc34667f82699d5d1ce1

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java (hunk 3 above)
Comment (GANGANI) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/impl/UserSessionManagementServiceImpl.java:95]: Fixed 8a7f2322bf7980e1b678bc34667f82699d5d1ce1

**Pull request merged**